DistroLauncher/AppConfig.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Oobe
+{
+    struct AppConfig
+    {
+        // The file name (including extension) of the executable that must be invoked as the OOBE.
+        const wchar_t* appName;
+
+        // Whether or not the fallback method should be skipped on OOBE failure.
+        bool mustSkipFallback;
+
+        // Whether the child process requires its own console or not.
+        bool requiresNewConsole;
+    };
+
+    // Returns the global constant instance of the AppConfig.
+    AppConfig appConfig();

```

Comment:
```suggestion
    struct AppConfig
    {
        // The file name (including extension) of the executable that must be invoked as the OOBE.
        constexpr std::wstring_view appName;

        // Whether or not the fallback method should be skipped on OOBE failure.
        constexpr bool mustSkipFallback;

        // Whether the child process requires its own console or not.
        constexpr bool requiresNewConsole;
    };

    // Returns the global constant instance of the AppConfig.
    constexpr AppConfig appConfig();
```
We're not very performance-blocked but might as well. Might even want to move the definition of `appConfig` to this header file to go all-in and make it trully constexpr.

Reply:
Not a big fan of making all data members const(expr) when we can make the whole object const(expr) from the outside. Not in this case, but in other cases that could hurt testability (thinking of the const-ness).

Also, I'd prefer to avoid showing #ifdef's in the header files. Following this premise require the function to be defined in the cpp, thus it would never be truly constexpr.

But the function being a singleton provider, it makes sense to make it return a const ref instead.

Reply:
Makes sense to me. However I believe that moving it to the sourcefile makes it not be constexpr (constexpr definitions need be in the same compilation unit). However this is very simple so LTO may even optimize it for us.